addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/messages/AbstractDevice.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.netatmo.internal.messages;
+
+import java.math.BigDecimal;
+import java.util.Calendar;
+import java.util.List;
+
+import org.apache.commons.lang.builder.ToStringBuilder;
+import org.codehaus.jackson.annotate.JsonAutoDetect;
+import org.codehaus.jackson.annotate.JsonAutoDetect.Visibility;
+import org.codehaus.jackson.annotate.JsonIgnoreProperties;
+import org.openhab.binding.netatmo.handler.AbstractEquipment.MeasureValueMap;
+
+/**
+ * Java Bean to represent a JSON common response elements of both modules and a
+ * devices
+ *
+ * @author Gaël L'hopital
+ */
+@JsonAutoDetect(fieldVisibility = Visibility.ANY)
+@JsonIgnoreProperties(ignoreUnknown = false)
+public abstract class AbstractDevice extends AbstractMessage {
+
+    protected String _id;
+    protected Integer firmware;
+    protected String module_name;
+    protected Boolean public_ext_data;
+    protected String type;
+    protected List<String> data_type;
+
+    // protected MeasureValueMap measures;
+
+    public Calendar timeStamp;
+
+    public String getId() {
+        return this._id;
+    }
+
+    public Integer getFirmware() {
+        return this.firmware;
+    }
+
+    public String getModuleName() {
+        return this.module_name;
+    }
+
+    public Boolean isPublicData() {
+        return this.public_ext_data;
+    }
+
+    public String getType() {
+        return this.type;
+    }
+
+    public List<String> getMeasurements() {
+        return this.data_type;
+    }
+
+    public Calendar gettimestamp() {
+        return timeStamp;
+    }
+
+    public BigDecimal getComputed(String chanel, MeasureValueMap measures) {
```

Comment:
Totally agree, in OH1 version I pushed this calculations in the action package of the Weather binding. I think it should be the same in the future with OH2 but it currently is not available